project_management_system/src/app/board-page/edit-task/edit-task.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+interface TaskData {
```

Comment:
It better to put interfaces in the separate file. It would be easy to import them, means import from some file not from component. Also, all interfaces will be cleaned up during the build but your components still will be more clean.